lib/galaxy/util/user_agent.py

```diff
@@ -0,0 +1,34 @@
+import urllib
+
+import requests
+
+
+def __append_word_to_user_agent(word):
+
+    # set requests User-Agent
+    old_default_user_agent = requests.utils.default_user_agent
+
+    def new_default_user_agent(*args):
+        return f"{old_default_user_agent(*args)} {word}"
+
+    requests.utils.default_user_agent = new_default_user_agent
+
+    # set urllib User-Agent
+    old_build_opener = urllib.request.build_opener
+
+    def modify_user_agent_header(header):
+        if header[0].lower() == "user-agent":
+            return (header[0], f"{header[1]} {word}")
+        return header
+
+    def new_build_opener(*handlers):
+        opener = old_build_opener(*handlers)
+        opener.addheaders = [
+            modify_user_agent_header(header) for header in opener.addheaders
+        ]
+        return opener
+
+    urllib.request.build_opener = new_build_opener
+
+
+__append_word_to_user_agent("galaxy")
```

Comment:
I think we could also stand to include the galaxy version here as well

Reply:
> I think we could also include the galaxy version here as well

Good idea, I added the galaxy version to the User-Agent.

Now, the primary outstanding TODO is to figure out where to import the `user_agent` package to get full coverage.  I probably should defer to y'all on that one, there's probably a best place that those with deeper knowledge of the codebase would know.

Reply:
Hello @hexylena, I'm checking in on this PR.  I think it's mostly done, except that someone with deep knowledge of the codebase (probably a regular contributor such as yourself) needs to determine where to import the `user_agent` package so that the user agent is properly set at all entry points.  Please let me know if there's any way I can help.